src/transformStateWithClones.js

```diff
@@ -6,8 +7,69 @@
  *
  * @return {Object[]}
  */
+// function transformStateWithClones(state, actions) {
+//   // write code here
+//   const table = {
+//     ...state,
+//   };
+//   const result = [];
+
+//   for (const action of actions) {
+//     if (action.type === 'clear') {
+//       for (const delet in table) {
+//         delete table[delet];
+//       }
+//       console.log(table);
+//     } else if (action.type === 'removeProperties') {
+//       for (const remove of action.keysToRemove) {
+//         delete table[remove];
+//       }
+//     } else {
+//       Object.assign(table, action.extraData);
+//     }
+
+//     result.push({ ...table });
+//   }
+
+//   return result;
+// }
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = {
+    ...state,
+  }
+
+  for (const action of actions) {
+    let copyOfState = {
+      ...currentState,
+    };
```

Comment:
[NAMING] - The variable name `copyOfState` is not descriptive enough. Consider renaming it to `stateCopy` or `currentStateCopy` to better convey its purpose.